Pharmacy/PharamcyApi/Controllers/MedicineProcurementController.cs

```diff
@@ -23,24 +24,46 @@ public MedicineProcurementController(IUnitOfWork uow)
         }
 
         [HttpPost("check")]
-        public IActionResult CheckMedicineAvailability(CheckMedicineAvailabilityRequestDTO checkMedicineAvailabilityRequestDTO)
+        public IActionResult CheckMedicineAvailability(CheckMedicineAvailabilityRequestDTO requestDTO)
         {
             try
             {
-                _hospitalAuthService.ValidateApiKey(checkMedicineAvailabilityRequestDTO.ApiKey);
-                bool isMedicineAvailable = _medicineProcurementService
-                    .IsMedicineAvailable(checkMedicineAvailabilityRequestDTO.MedicineName,
-                    checkMedicineAvailabilityRequestDTO.ManufacturerName, checkMedicineAvailabilityRequestDTO.Quantity);
+                _hospitalAuthService.ValidateApiKey(requestDTO.ApiKey);
+                bool answer = _medicineProcurementService.IsMedicineAvailable(requestDTO.MedicineName, requestDTO.ManufacturerName, requestDTO.Quantity);
 
-                return Ok(new CheckMedicineAvailabilityResponseDTO() { Answer = isMedicineAvailable });
+                return Ok(new CheckMedicineAvailabilityResponseDTO() { Answer = answer });
             }
-            catch (UnauthorizedAccessException uae)
+            catch (UnauthorizedAccessException exception)
             {
-                return Unauthorized(uae.Message);
+                return Unauthorized(exception.Message);
             } 
-            catch (KeyNotFoundException knfe)
+            catch (MedicineFromManufacturerNotFoundException exception)
             {
-                return NotFound(knfe.Message);
+                return NotFound(exception.Message);
+            }
+        }
+
+        [HttpPost("execute")]
+        public IActionResult ExecuteMedicineProcurement(MedicineProcurementRequestDTO requestDTO)
+        {
```

Comment:
What if some other exception apart from these occur? These are "expected possible mistakes", but we don't know if something else happens what will handle that. Probably would add one more catch for general exception where i would return Problem(e.Message)